src/main/java/net/minecraftforge/common/ForgeConfigSpec.java

```diff
@@ -741,6 +838,30 @@ public boolean test(Object t)
             return result;
         }
 
+        public Component getErrorMessage(Object t) {
+            if (isNumber(t))
+            {
+                Number n = (Number) t;
+                boolean result = ((Number)min).doubleValue() <= n.doubleValue() && n.doubleValue() <= ((Number)max).doubleValue();
+                if(!result)
+                {
+                    return Component.translatable("forge.configgui.error.ranged.notInBounds", n.doubleValue(), ((Number)min).doubleValue(), ((Number)max).doubleValue());
+                }
+
+                throw new IllegalStateException("Called the error message producor for a valid value!");
+            }
+            if (!clazz.isInstance(t)) return Component.translatable("forge.configgui.error.ranged.needsToBeOfType", clazz.getSimpleName());
+            V c = clazz.cast(t);
+
+            boolean result = c.compareTo(min) >= 0 && c.compareTo(max) <= 0;
```

Comment:
I wonder if instead of having an inverted if statement below we should just have something along the lines of:
```java
if (check from this line)
throw exception valid value
```